src/main/java/org/eclipse/uprotocol/transport/builder/UAttributesBuilder.java

```diff
@@ -48,73 +49,83 @@ public class UAttributesBuilder {
 
     /**
      * Construct a UAttributesBuilder for a publish message.
+     * @param source   Source address of the message.
      * @param priority The priority of the message.
      * @return Returns the UAttributesBuilder with the configured priority.
      */
-    public static UAttributesBuilder publish(UPriority priority) {
+    public static UAttributesBuilder publish(UUri source, UPriority priority) {
+        Objects.requireNonNull(source, "source cannot be null.");
         Objects.requireNonNull(priority, "UPriority cannot be null.");
-        return new UAttributesBuilder(UuidFactory.Factories.UPROTOCOL.factory().create(), 
+        return new UAttributesBuilder(source, UuidFactory.Factories.UPROTOCOL.factory().create(), 
         UMessageType.UMESSAGE_TYPE_PUBLISH, priority);
     }
 
 
     /**
      * Construct a UAttributesBuilder for a notification message.
+     * @param source   Source address of the message.
      * @param priority The priority of the message.
      * @param sink The destination URI.
      * @return Returns the UAttributesBuilder with the configured priority and sink.
      */
-    public static UAttributesBuilder notification(UPriority priority, UUri sink) {
+    public static UAttributesBuilder notification(UUri source, UPriority priority, UUri sink) {
```

Comment:
Can we re-order arguments: source, sink, priority... (Here and other factory methods)

Reply:
Done.